### PR TITLE
Add spider for João Pessoa/PB

### PR DESCRIPTION
**Pull request by alfakini**
Hey people,

In this PR we add the spider to collect data from [João Pessoa/PB gazette](http://www.joaopessoa.pb.gov.br/semanariooficial/).

### CODE REVIEW

File: processing/data_collection/gazette/spiders/pb_joao_pessoa.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+from datetime import datetime
+from scrapy import Request, Spider
+from gazette.items import Gazette
+from gazette.spiders.base import BaseGazetteSpider
+
+
+class PbJoaoPessoaSpider(BaseGazetteSpider):
+    TERRITORY_ID = "2507507"
+
+    EXTRA_EDITION_CSS = "td:first-child"
```
Comment (giovanisleite) [inline processing/data_collection/gazette/spiders/pb_joao_pessoa.py]: All selectors working :+1:

File: processing/data_collection/gazette/spiders/pb_joao_pessoa.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        @url http://www.joaopessoa.pb.gov.br/semanariooficial/
+        @returns requests 1
+        @scrapes date file_urls is_extra_edition territory_id power scraped_at
+        """
+
+        for element in response.css(self.GAZETTE_ROW_CSS):
+            url = element.css(self.GAZETTE_URL_CSS).extract_first()
+            date = element.css(self.DATE_CSS).re(self.DATE_REGEX).pop()
+            date = dateparser.parse(date, languages=["pt"]).date()
+            is_extra = "Especial" in element.css(self.EXTRA_EDITION_CSS).extract_first()
```
Comment (giovanisleite) [inline processing/data_collection/gazette/spiders/pb_joao_pessoa.py]: Extra edition identified successfully :+1:

File: processing/data_collection/gazette/spiders/pb_joao_pessoa.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+                date=date,
+                file_urls=[url],
+                is_extra_edition=is_extra,
+                territory_id=self.TERRITORY_ID,
+                power="executive_legislature",
+                scraped_at=datetime.utcnow(),
+            )
+
+        for url in response.css(self.NEXT_PAGE_CSS).extract():
+            yield Request(url)
```
Comment (giovanisleite) [inline processing/data_collection/gazette/spiders/pb_joao_pessoa.py]: Working :+1: 

just a good to know note, a shortcut to create a request that I learned from rennerocha it is ```response.follow```

File: processing/data_collection/gazette/spiders/pb_joao_pessoa.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    DATE_REGEX = r"[0-9]{2}/[0-9]{2}/[0-9]{4}"
+
+    allowed_domains = ["www.joaopessoa.pb.gov.br"]
+    name = "pb_joao_pessoa"
+    start_urls = ["http://www.joaopessoa.pb.gov.br/semanariooficial/"]
+
+    def parse(self, response):
+        """
+        @url http://www.joaopessoa.pb.gov.br/semanariooficial/
+        @returns requests 1
```
Comment (giovanisleite) [inline processing/data_collection/gazette/spiders/pb_joao_pessoa.py]: What about test against the items?

File: processing/data_collection/gazette/spiders/pb_joao_pessoa.py [hunk 5]
```diff
@@ -0,0 +1,46 @@
+import dateparser
+
+from datetime import datetime
+from scrapy import Request, Spider
```
Comment (giovanisleite) [inline processing/data_collection/gazette/spiders/pb_joao_pessoa.py]: Spider imported but never used

**Comment by alfakini** [comment]
Thanks for the review @giovanisleite! I fixed the issues you raises :)

**Comment by phrfpeixoto** [comment]
Linking with #186

**Comment by giuliocc** [comment]
#186 was almost working, just needed small adjustments. Since it was stale for a long time #270 was opened.

**Comment by jvanz** [comment]
Closing this in favor of #270

**Pull request closed without merging**